templates/apigee-entity-default.html.twig

```diff
@@ -0,0 +1,80 @@
+{#
+/**
+ * @file
+ * Default theme implementation to display an Apigee entity.
+ *
+ * @see \Drupal\apigee_edge\Entity\EdgeEntityViewBuilder::getBuildDefaults()
+ * @see \Drupal\Core\Entity\EntityViewBuilder::getBuildDefaults()
+ *
+ * If you need an entity type specific template, do not override this.
+ * Register an entity type specific theme in hook_theme() and implement your
+ * own template_preprocess_ENTITY_TYPE() template preprocess, template file and
+ * hook_theme_suggestions_ENTITY_TYPE() if needed.
+ *
+ * @code
+ *   function MY_MODULE_theme() {
+ *     return [
+ *       'developer_app' => [
+ *          'render element' => 'elements',
+ *       ],
+ *     ];
+ *   }
+ *
+ *   function template_preprocess_developer_app(&$variables) {};
+ *
+ *   function MY_MODULE_theme_suggestions_developer_app(array $variables) {}
+ * @endcode
+ *
+ * If you would like to re-use the implementation from
+ * template_preprocess_apigee_entity_default() you can call this function in
+ * your own entity type specific template preprocess function or add it to the
+ * entity type specific preprocess functions in hook_theme_registry_alter().
```

Comment:
Everything from line 6-31 should not be here. We should not have to educate on how to create theme hooks, base preprocess hooks, or have class references, especially not in a Twig template, you know?  This is not worth the perceived gain of having one vs multiple templates IMO.

Reply:
I always appreciate more documentation than less, but that does seem like a problem if we need to explain how our patterns work versus already established conventions.

Reply:
@jacine I felt necessary to educate developers who are going to customize the look and feel of the Apigee entities because if the fix for the https://www.drupal.org/project/drupal/issues/2808481 issue lands in Drupal core it will create a confusing situation.

There will be multiple themes and templates that could be used to theme an entity, for example, a developer app and you must know the order how Drupal will try to apply them:
* If you define the `developer_app` theme in `hook_theme()` on your module, this is going to be the primary theme for the developer_app entity and it always gets chosen by the EntityViewBuilder. The related template that you use is `developer-app.html.twig`. (Just like now)
See:
https://github.com/drupal/core/blob/17557c97a816bc78e70989a86064c562bc27027f/lib/Drupal/Core/Entity/EntityViewBuilder.php#L174-L177
* If that theme does not exist, Drupal will try to apply the default "entity" theme and because of that a developer app's view could be customized`entity--developer-app.twig.html` template.

This is two theme and multiple templates already that you can possibly use to customize one entity's view.

I did not want to create another with the "apigee_entity_default" theme. It is just a fallback solution to fix a Drupal issue that causing problems for us and it seems it won't be fixed till Drupal 8.8. This is the reason why I am trying to discourage everybody to try to implement something like this: `apigee-entity-developer-app.html.twig`. (Originally, I implemented it a way like this.)

So with this, I am trying to avoid confusion and educate developers who may not know what EntityViewBuilder does and will do in the background if the fix lands in Drupal core. This is the reason why I left this comment. Everybody who would like to customize the look and feel of an entity should define the entity type specific theme and template because that is the one that always gets applied by EntityViewBuilder if exists.